deps/src/polymake_functions.cpp

```diff
@@ -19,9 +22,63 @@ void initialize_polymake()
     }
 }
 
+polymake::perl::Object call_func_0args(std::string func)
+{
+    return polymake::call_function(func);
+}
+
+polymake::perl::Object call_func_1args(std::string func, int arg1)
+{
+    return polymake::call_function(func, arg1);
+}
+
+polymake::perl::Object call_func_2args(std::string func, int arg1, int arg2)
+{
+    return polymake::call_function(func, arg1, arg2);
+}
+
 pm::perl::Object to_perl_object(pm::perl::PropertyValue v)
 {
     pm::perl::Object obj;
     v >> obj;
     return v;
 }
+
+std::string typeinfo_helper(pm::perl::PropertyValue p, bool demangle)
+{
+    PropertyValueHelper ph(p);
+
+    if (!ph.is_defined()) {
+        return "undefined";
+    }
+    if (ph.is_boolean()) {
+        return "bool";
+    }
+    switch (ph.classify_number()) {
+        // primitives
+        case PropertyValueHelper::number_is_zero:
+        case PropertyValueHelper::number_is_int:
+            return "int";
+        case PropertyValueHelper::number_is_float:
+            return "double";
+
+        // with typeinfo ptr (nullptr for Objects)
+        case PropertyValueHelper::number_is_object:
+            // some non-primitive Scalar type with typeinfo (e.g. Rational)
+        case PropertyValueHelper::not_a_number:
+            // a c++ type with typeinfo or a perl Object
+            {
+                const std::type_info* ti = ph.get_canned_typeinfo();
+                if (ti == nullptr) {
+                    return "perl::Object";
```

Comment:
The string here does not contain the ```pm::```, adding it here should avoid the special case in the ```WrappedTypes```.

Also, when there is an object of type ```pm::perl::Object``` inside a ```PropertyValue``` there is no ```type_info``` struct, and we currently just guess this to be a ```pm::perl::Object```. (It might also be: pm::Array<pm::perl::Object>, pm::perl::ObjectType or some plain perl type, and maybe more...)
I'm not sure if there is a way to determine these types at runtime, we might need to use/trust the json for those.